src/blocks/memory.rs

```diff
@@ -271,12 +313,52 @@ impl Memstate {
                 "SwapTotal:" => mem_state.swap_total = val,
                 "SwapFree:" => mem_state.swap_free = val,
                 "SwapCached:" => mem_state.swap_cached = val,
+                "Zswap:" => mem_state.zswap_compressed = val,
+                "Zswapped:" => mem_state.zswap_decompressed = val,
                 _ => (),
             }
 
             line.clear();
         }
 
+        // For ZRAM
+        let mut entries = read_dir("/sys/block/")
+            .await
+            .error("Could not read /sys/block")?;
+        while let Some(entry) = entries
+            .next_entry()
+            .await
+            .error("Could not get next file /sys/block")?
+        {
+            if let Some(name) = entry.file_name().to_str() {
```

Comment:
To make it slightly less nested: `let Some(name) = entry.file_name().to_str() else { continue };`